r/R/record-batch.R

```diff
@@ -189,3 +189,59 @@ record_batch <- RecordBatch$create
 
 #' @export
 names.RecordBatch <- function(x) x$names()
+
+#' @export
+rbind.RecordBatch <- function(...) {
+  abort("Use `Table$create()` to combine record batches")
```

Comment:
Is this better (more explicit) or worse (confusing because it's telling me I'm not going to get a RecordBatch)?

Is there or should there be a ConcatenateBatches function, and/or some way to turn a Table into a RecordBatch (concat_arrays each of the ChunkedArrays in the Table)? 

```suggestion
  abort("Use `Table$create()` to combine RecordBatches into a Table")
```

Reply:
This is a similar situation to `c.Array`, where there's a zero-copy solution that gets you a chunked array and a solution that copies to give you an Array. I was thinking that we'd rather the user made an explicit choice about what they want, rather than make that for them. Though we don't have a `concat_record_batches` function here yet, we would present that as another option here if we did.

Reply:
Yeah I agree that you should make it an explicit choice. Just noting the difference between this and `c.Array`, which give you options.